moto/awslambda/models.py

```diff
@@ -1276,7 +1276,7 @@ def update_from_cloudformation_json(  # type: ignore[misc]
         cloudformation_json: Dict[str, Any],
         account_id: str,
         region_name: str,
-    ) -> "EventSourceMapping":
+    ) -> "Optional[EventSourceMapping]":
```

Comment:
We'd get none if the json dict passed in tried to create an event source mapping for a lambda function that doesn't actually exist.

making this Optional fixes the issue, but maybe the None path should raise an error instead? I'm not sure how moto generally handles this type of situation

Reply:
It should raise an error, ideally. There will be lots of scenarios like this, where we should validate how AWS behaves and then reproduce it in Moto.

Considering the effort involved, I'd vote to just `type: ignore` it for now. If/when someone runs into this error, we can always have another look. Same goes for the other scenarios.